### PR TITLE
Testing PIXI.js on GameTest Page

### PR DESCRIPTION
**Pull request by yummyblabla**
- [x] Create Sprite Factory
- [x] Create Interaction Factory (not necessary for now as we only need mouse interaction, only contains static function)
- [x] Align PIXI Application to whole browser screen (possibly responsive to screen size changes)
- [x] Create classes to hold game related sprites into PIXI containers

### CODE REVIEW

File: src/modules/mahjong/MahjongPlayer/MahjongPlayer.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    const container = new PIXI.Container();
+    if (this.hand !== undefined) {
+      const selectedTile = this.hand.getSelectedTile();
+      const tiles = this.hand.getHand();
+
+      tiles.forEach((tile, index) => {
+        const frontSprite = spriteFactory.generateSprite(FRONT_TILE);
+        frontSprite.width = DEFAULT_MAHJONG_WIDTH;
+        frontSprite.height = DEFAULT_MAHJONG_HEIGHT;
+        frontSprite.x = index * (DEFAULT_MAHJONG_WIDTH + DISTANCE_FROM_TILES);
```
Comment (jzc1828) [inline src/modules/mahjong/MahjongPlayer/MahjongPlayer.ts:58]: can set an initial x value to 70-80,
```js
const INITIAL_SPRITE_X = 80
frontSprite.x = INITIAL_SPRITE_X + index * (DEFAULT_MAHJONG_WIDTH + DISTANCE_FROM_TILES);
```

File: src/modules/mahjong/MahjongPlayer/MahjongPlayer.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        container.addChild(frontSprite);
+
+        const sprite = spriteFactory.generateSprite(tile.toString());
+        sprite.width = DEFAULT_MAHJONG_WIDTH;
+        sprite.height = DEFAULT_MAHJONG_HEIGHT;
+        sprite.x = index * (DEFAULT_MAHJONG_WIDTH + DISTANCE_FROM_TILES);
```
Comment (jzc1828) [inline src/modules/mahjong/MahjongPlayer/MahjongPlayer.ts:64]: can set an initial x value to 70-80,
```js
const INITIAL_SPRITE_X = 80
sprite.x = INITIAL_SPRITE_X + index * (DEFAULT_MAHJONG_WIDTH + DISTANCE_FROM_TILES);
```

File: src/pages/Game.tsx [hunk 3]
```diff
@@ -1,12 +1,230 @@
...
+   * @param payload GamePageLoadJSON
+   */
+  const confirmGamePageLoadReceived = (payload: unknown): void => {
+    const data = payload as GamePageLoadJSON;
+    const { success, error } = data;
+    if (success) {
+      ws?.removeListener(IncomingAction.GAME_PAGE_LOAD);
+    } else {
+      console.log(`Error processing GAME_PAGE_LOAD: ${error}`);
+      ws?.sendMessage(OutgoingAction.GAME_PAGE_LOAD, { success: true });
```
Comment (jzc1828) [inline src/pages/Game.tsx:134]: do you mean `success: false`?
```js
ws?.sendMessage(OutgoingAction.GAME_PAGE_LOAD, { success: false });
```

File: src/pages/Game.tsx [hunk 4]
```diff
@@ -1,12 +1,230 @@
...
+     */
+    interactionManager = pixiApplication.renderer.plugins.interaction;
+    pixiLoader = pixiApplication.loader;
+    console.log(interactionManager);
+    stage = pixiApplication.stage;
+
+    /**
+     * Function setup invoked when assets are done loading
+     */
+    function setup(loader: PIXI.Loader, resources: Partial<Record<string, PIXI.LoaderResource>>) {
```
Comment (jzc1828) [inline src/pages/Game.tsx:179]: missing void as function return type

**Comment by jzc1828** [comment]
currently, I am experiencing errors coming from the backend when I clicked start game in the frontend:
error msg:
```
{"message":"The number of conditions on the keys is invalid","code":"ValidationException","time":"2020-07-23T17:17:42.855Z","requestId":"af4dbe7a-ed9e-4570-ae12-9a65c82b99c5","statusCode":400,"retryable":false,"retryDelay":37.8049920434904}
```

File: src/pages/Game.tsx (hunk 3 above)
Comment (yummyblabla) [reply src/pages/Game.tsx:134]: Does the backend care that the payload: success is true or false

Comment (jzc1828) [reply src/pages/Game.tsx:137]: actually I need `gameId` as the payload for `GAME_PAGE_LOAD`
```
ws?.sendMessage(OutgoingAction.GAME_PAGE_LOAD, { gameId: game.gameId });
```

File: src/pages/Game.tsx [hunk 5]
```diff
@@ -1,12 +1,233 @@
...
+    console.log(interactionManager);
+    stage = pixiApplication.stage;
+
+    /**
+     * Function setup invoked when assets are done loading
+     */
+    function setup(loader: PIXI.Loader, resources: Partial<Record<string, PIXI.LoaderResource>>): void {
+      console.log(loader);
+      if (ws) {
+        ws.sendMessage(OutgoingAction.GAME_PAGE_LOAD, { success: true });
```
Comment (jzc1828) [inline src/pages/Game.tsx:185]: same here:
```js
ws.sendMessage(OutgoingAction.GAME_PAGE_LOAD, { gameId: game.gameId });
```

File: src/modules/ws/message_validator.ts [hunk 6]
```diff
@@ -17,6 +17,7 @@ const MESSAGE_FORMATS: Record<string, string[]> = {
   [OutgoingAction.JOIN_GAME]: ['gameId'],
   [OutgoingAction.LEAVE_GAME]: ['gameId'],
   [OutgoingAction.START_GAME]: ['gameId'],
+  [OutgoingAction.GAME_PAGE_LOAD]: ['success'],
```
Comment (jzc1828) [inline src/modules/ws/message_validator.ts:20]: same here:
```js
[OutgoingAction.GAME_PAGE_LOAD]: ['gameId'],
```

File: src/pages/Game.tsx (hunk 3 above)
Comment (jzc1828) [reply src/pages/Game.tsx:137]: and these payloads for these two routes
```
DRAW_TILE: { gameId: 'game Id' }
PLAY_TILE: { gameId: 'game Id', tile: '1_DOT' }
```

Comment (yummyblabla) [reply src/pages/Game.tsx:137]: naming convention is same for inbound and outbound messages?
e.g. FrontEnd -> DRAW_TILE -> Backend -> DRAW_TILE -> FrontEnd

File: src/pages/Game.tsx (hunk 5 above)
Comment (yummyblabla) [reply src/pages/Game.tsx]: Done

File: src/modules/ws/message_validator.ts (hunk 6 above)
Comment (yummyblabla) [reply src/modules/ws/message_validator.ts]: Done

File: src/pages/Game.tsx (hunk 3 above)
Comment (yummyblabla) [reply src/pages/Game.tsx]: Nvm, I see it on the updated UML diagram

Comment (jzc1828) [reply src/pages/Game.tsx]: yes

**Pull request merged**